web.py

```diff
@@ -172,6 +175,39 @@ async def root():
         """
 
 
+class FormattedFloat(float):
+    def __repr__(self):
+        return '{:.18f}'.format(self).rstrip('0').rstrip('.')
+
+
+class JsonFloatEncoder(json.JSONEncoder):
+    def encode(self, val):
+        if isinstance(val, dict):
+            return {k: self.encode(v) for k, v in val.items()}
+
+        if isinstance(val, (list, tuple)):
+            return type(val)(self.encode(v) for v in val)
+
+        if isinstance(val, float):
+            return FormattedFloat(val)
+
+        return val
+
+
+class FormattedJSONResponse(JSONResponse):
+    media_type = "application/json"
+
+    def render(self, content: typing.Any) -> bytes:
+        return json.dumps(
+            str(content).replace("'", "\"").replace(", ", ",").replace(": ", ":"),
```

Comment:
Why do you need all these replace? I don't think they break JSON.

Reply:
It could break for some parsers